crates/sp-lightclient/src/lib.rs

```diff
@@ -317,4 +318,79 @@ impl<Header: HeaderT, Store: Storage<Header>> HeaderImporter<Header, Store> {
         let tag = u64::from_be_bytes(pre_digest.solution.tag);
         u128::from(u64::MAX - subspace_core_primitives::bidirectional_distance(&target, &tag))
     }
+
+    /// Returns the ancestor of the header at number.
+    fn find_ancestor_of_header_at_number(
+        store: &Store,
+        header: HeaderExt<Header>,
+        ancestor_number: NumberOf<Header>,
+    ) -> Option<HeaderExt<Header>> {
+        // header number must be greater than the ancestor number
+        if ancestor_number >= *header.header.number() {
+            return None;
+        }
+
+        let headers_at_ancestor_number = store.headers_at_number(ancestor_number);
+        let finalized_header = store.finalized_header();
+
+        // short circuit if the ancestor number is at the same or lower number than finalized head
+        if ancestor_number.le(finalized_header.header.number())
+            // short circuit if there are no forks at the depth
+            || headers_at_ancestor_number.len() == 1
```

Comment:
I think above condition is redundant, we must not ever have more than one block at or below finalization point, so check for one ancestor is sufficient, we don't need to check height above.